File: verification/verify_keys.js
const { chromium } = require('@playwright/test');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Wait for the app to start
  console.log('Waiting for app to start...');

  // Retry connection a few times
  for (let i = 0; i < 30; i++) {
    try {
      await page.goto('http://localhost:3000');
      break;
    } catch (e) {
      console.log('Retrying connection...');
      await new Promise(resolve => setTimeout(resolve, 1000));
    }
  }

  // Check if "Mission Brief" is visible
  try {
    const brief = page.locator('text=Mission Brief');
    await brief.waitFor({ timeout: 10000 });
    console.log('Mission Brief found.');
  } catch (e) {
    console.log('Mission Brief NOT found, taking screenshot anyway.');
  }

  // Take a screenshot of the Mission Brief
  await page.screenshot({ path: 'verification/mission_brief_keys.png' });
  console.log('Screenshot saved to verification/mission_brief_keys.png');

  await browser.close();
})();

File: verification/verify_keys.py
from playwright.sync_api import sync_playwright

def verify_keys(page):
    page.goto("http://localhost:3000")

    # Wait for the mission brief modal
    page.wait_for_selector('text="Mission Brief"')

    # Take a screenshot of the Mission Brief
    page.screenshot(path="verification/mission_brief_keys.png")

    # Verify some keys exist
    # Since we replaced text with visual keys, we can check for the element or just rely on the screenshot
    # But let's check if the text "W" exists (it's inside the key)
    assert page.get_by_text("W", exact=True).is_visible()

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        try:
            verify_keys(page)
        except Exception as e:
            print(e)
        finally:
            browser.close()
